components/HomeProducts.tsx

```diff
@@ -0,0 +1,72 @@
+import { useState, useEffect} from 'react'
+import { Flex, Grid, Box, Text} from '@chakra-ui/react'
+import Product from './Product'
+
+function HomeProducts({coffees, itemsLeft}) {
+  const [coffeesArr, setCoffeeArr] = useState([...coffees])
+  const [coffeesLeft, setCoffeesLeft] = useState(itemsLeft)
+  const [query, setQuery] = useState(false)
+
+  useEffect( ()=> {
+    if (!query) return
+
+    const fetchData = async () => {
+      const res = await fetch("http://localhost:3000/api/hello")
+      const data = await res.json()
+
+      const newCoffees = await data.slice(9)
+      const isCoffees= coffeesLeft - newCoffees.length
```

Comment:
well, there is a different way to do this, and the hint is, if the backend doesnt give you more products, that means that you don't need to show more products.